<a href="https://colab.research.google.com/github/gredy/2021Z-DataVisualizationTechniques/blob/master/garrido_project_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# PROJECT -BREAST-CANCER-GARRIDO


In [21]:
# DATA SET

import pandas as pd
import numpy as np
from sklearn import datasets

# Load the dataset
data = datasets.load_breast_cancer()

# Create a Pandas DataFrame
df = pd.DataFrame(data.data, columns=data.feature_names)
df['target'] = data.target

# Display the first few rows of the dataset
df.head()

mean radius  mean texture  mean perimeter  ...  worst symmetry  worst fractal dimension  target
0        17.99         10.38          122.80  ...          0.4601                  0.11890       0
1        20.57         17.77          132.90  ...          0.2750                  0.08902       0
2        19.69         21.25          130.00  ...          0.3613                  0.08758       0
3        11.42         20.38           77.58  ...          0.6638                  0.17300       0
4        20.29         14.34          135.10  ...          0.2364                  0.07678       0

[5 rows x 31 columns]

In [22]:
#LIBRARIES  OF THE CLASSIFIERS
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import LabelEncoder
from sklearn.metrics import accuracy_score
from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.svm import SVC
from sklearn.linear_model import Perceptron
from sklearn.linear_model import SGDClassifier
from sklearn.neighbors import KNeighborsClassifier
from sklearn.preprocessing import StandardScaler



# Prepare the data
X = df.drop(['target'], axis=1)  # Feature columns (excluding the target column)
y = df["target"]  # Target column

# Encode target variable
label_encoder = LabelEncoder()
y_encoded = label_encoder.fit_transform(y)  # 'B' -> 0, 'M' -> 1

# Split the dataset into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y_encoded, test_size=0.2, random_state=42)

# Standardize features
scaler = StandardScaler()
X_train = scaler.fit_transform(X_train)
X_test = scaler.transform(X_test)

# Initialize classifiers
classifiers = {
    "Logistic Regression": LogisticRegression(max_iter=1000),
    "Decision Tree": DecisionTreeClassifier(),
    "Random Forest": RandomForestClassifier(n_estimators=100),
    "Support Vector Machine": SVC(),
     "perceptron": Perceptron(),
      "SGD Classifier": SGDClassifier(),
      "knn": KNeighborsClassifier(n_neighbors=5,p=2,metric='minkowski')
    }

# Train and evaluate classifiers
accuracy_results = {}
for name, clf in classifiers.items():
    clf.fit(X_train, y_train)
    y_pred = clf.predict(X_test)
    accuracy_results[name] = accuracy_score(y_test, y_pred)


# Display the accuracy results
import pandas as pd
from IPython.display import display

# Display the accuracy results
accuracy_df = pd.DataFrame(list(accuracy_results.items()), columns=["Algorithm", "Accuracy"])
#print(accuracy_df)  # Or display(accuracy_df) for a formatted output

# Use display() to show the DataFrame
display(accuracy_df)

Algorithm  Accuracy
0     Logistic Regression  0.973684
1           Decision Tree  0.938596
2           Random Forest  0.964912
3  Support Vector Machine  0.982456
4              perceptron  0.973684
5          SGD Classifier  0.964912
6                     knn  0.947368

# Conclusion
1. all the algorithms are greater than .94
2. Because SVM - Accuracy is reporting .982, it is the one to be submitted and explored.

# CREATING THE PREDICTIONS
That section We compute the prediction using the SVM model of Classification


In [13]:
# SVM MODEL
from sklearn.metrics import accuracy_score
from sklearn.linear_model import SGDClassifier
svm=SGDClassifier(loss='hinge')
svm.fit(X_train,y_train)
y_pred=svm.predict(X_test)
accuracy_score(y_test,y_pred)
print('Accuracy: %.3f',accuracy_score(y_test,y_pred))

Accuracy: %.3f 0.9736842105263158


In [20]:
# SVM MODEL
from sklearn.metrics import accuracy_score
from sklearn.linear_model import SGDClassifier
svm=SGDClassifier(loss='log_loss')
modelt=svm.fit(X_train,y_train)
y_pred=svm.predict(X_test)
accuracy_score(y_test,y_pred)
print(accuracy_score(y_test,y_pred))


0.9912280701754386


SGDClassifier(loss='log_loss')